### PR TITLE
Config mnesia setup and replication

### PR DESCRIPTION
**Pull request by norbajunior**

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -2,7 +2,7 @@ import Config
 
 config :rinha,
   port: System.get_env("PORT", "4000"),
-  nodes: [:rinha@api01, :rinha@api02]
+  nodes: [:"rinha1@Norbertos-MacBook-Pro-3", :"rinha2@Norbertos-MacBook-Pro-3"]
```
Comment (norbajunior) [inline config/config.exs]: Mudar pra localhost aqui

File: lib/rinha/database.ex [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    else
+      error ->
+        Logger.error("Error configuring mnesia nodes: #{inspect(error)}")
+        :ok
+    end
+  end
+
+  defp create_schema(node) do
+    :mnesia.delete_schema([node])
+
```
Comment (norbajunior) [inline lib/rinha/database.ex]:
```suggestion
```

File: lib/rinha/database.ex [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        Logger.info("transactions table already exists")
+        :ok
+
+      {:aborted, {:bad_type, Transaction, :disc_only_copies, _}} ->
+        Logger.info("transactions table already exists")
+        :ok
+
+      {:aborted, {:not_active, Transaction, _}} ->
+        Logger.info("customers table already exists")
+        :ok
```
Comment (norbajunior) [inline lib/rinha/database.ex]:
```suggestion
```

File: lib/rinha/database.ex [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+      {:aborted, {:bad_type, Customer, :disc_only_copies, _}} ->
+        Logger.info("customers table already exists")
+        :ok
+
+      {:aborted, {:not_active, Customer, _}} ->
+        Logger.info("customers table already exists")
+        :ok
+
```
Comment (norbajunior) [inline lib/rinha/database.ex]:
```suggestion
```

File: lib/rinha/database.ex [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        Logger.info("transactions table already exists")
+        :ok
+
+      {:aborted, {:bad_type, Transaction, :disc_only_copies, _}} ->
+        Logger.info("transactions table already exists")
+        :ok
+
+      {:aborted, {:not_active, Transaction, _}} ->
+        Logger.info("customers table already exists")
+        :ok
```
Comment (norbajunior) [inline lib/rinha/database.ex]:
```suggestion
```

File: lib/rinha/node_monitor.ex [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  def start_link(_args) do
+    GenServer.start_link(__MODULE__, [], name: {:global, __MODULE__})
+  end
+
+  @impl true
+  def init(_args) do
+    :ok = :net_kernel.monitor_nodes(true)
+
+    if node() == @first_node do
+      Process.send_after(self(), :config_database, 500)
```
Comment (norbajunior) [inline lib/rinha/node_monitor.ex]:
```suggestion
      Process.send_after(self(), setup_database, 500)
```

File: lib/rinha/node_monitor.ex [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    end
+
+    {:ok, []}
+  end
+
+  @impl true
+  def handle_info(:config_database, state) do
```
Comment (norbajunior) [inline lib/rinha/node_monitor.ex]:
```suggestion
  def handle_info(:setup_database, state) do
```

File: lib/rinha/database.ex [hunk 8]
```diff
@@ -1,56 +1,74 @@
...
-           index: []
+           index: [],
+           disc_only_copies: [node()]
          ) do
       {:atomic, :ok} ->
         Logger.info("customers table has been created")
         :ok
 
-      {:aborted, {:already_exists, Customer}} ->
+      {:aborted, {_, Customer}} ->
```
Comment (norbajunior) [inline lib/rinha/database.ex]:
```suggestion
      {:aborted, {:already_exists, Customer}} ->
```

File: config/config.exs (hunk 1 above)
Comment (norbajunior) [inline config/config.exs]:
```suggestion
  nodes: [:"rinha1@localhost", :"rinha2@localhost"]
```

**Review by renanvy — APPROVED** [review]
👏🏽

**Pull request merged**